### PR TITLE
Update README with example of using Authorizer

### PR DESCRIPTION
**Pull request by jaypipes**
Adds documentation and examples of how to use the
external Authorizer plugin.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,7 +159,7 @@ Talons comes with a few simple examples of Authenticator plugins.
 ### `talons.auth.external.Authenticator`
 
 A generic Authenticator plugin that has one main configuration option,
-`authenticate_external_authfn` which should be the "module.function" or
+`external_authn_fn` which should be the "module.function" or
```
Comment (sebasmagri) [inline README.md]: Would this be external_auth_fn?

Comment (jaypipes) [inline README.md]: I kept the code backwards compatible with the original `external_authfn` config option:

https://github.com/jaypipes/talons/blob/master/talons/auth/external.py#L57

But this was supposed to be external_authn_callable. Doh. Updating now...

I needed to differentiate between authn and authz plugins...

File: README.md [hunk 2]
```diff
@@ -242,11 +242,78 @@ called an "Authorizer". Each Authorizer implements a single method,
...
+        m = p.match(res_string)
+        if m:
+            return fn(m, identity)
+```
+
+To use the above `application.auth.authorize` method for authorizing the
+identity that was authenticated, we'd supply the following configuration
+options to the `talons.auth.external.Authorizer` constructor:
+
+ * `external_authz_fn=application.auth.authorize`
```
Comment (queertypes) [inline README.md]: `external_authz_fn` is inconsistent with `external_authn_fn` above.

File: README.rst [hunk 3]
```diff
@@ -259,6 +259,90 @@ option:
...
+            m = p.match(res_string)
+            if m:
+                return fn(m, identity)
+
+To use the above ``application.auth.authorize`` method for authorizing
+the identity that was authenticated, we'd supply the following
+configuration options to the ``talons.auth.external.Authorizer``
+constructor:
+
+-  ``external_authz_fn=application.auth.authorize``
```
Comment (queertypes) [inline README.rst]: As noted in the markdown version of the README - inconsistent name error.

File: README.rst [hunk 4]
```diff
@@ -259,6 +259,90 @@ option:
...
    for authentication checks.
 
+Authorizers
+-----------
+
+Each class that derives from ``talons.auth.interfaces.Authorizes`` is
+called an "Authorizer". Each Authorizer implements a single method,
+``authorize()``, that takes a ``talons.auth.interfaces.Identity``
+object, a ``talons.auth.interfaces.ResourceAction`` object.
+
```
Comment (sebasmagri) [inline README.rst:269]: Would we replace that comma before 'a ``talons.auth...' with 'and'?

Comment (queertypes) [inline README.rst:269]: Yes, that's correct.

Each `Authorizer` implements a single method `authorize()` that takes a `talons.auth.interfaces.Identity` object **and** a `talons.auth.interfaces.ResourceAction` object.

**Pull request closed without merging**